Keas.Mvc/ClientApp/util/api.ts

```diff
@@ -16,7 +16,7 @@ const doFetch = async (
   });
 
   if (!res.ok) {
-    throw new Error(res.statusText);
+    throw await res.json();
```

Comment:
I don't see any reason to believe that the error result will be JSON in all cases.  In fact I think in many cases it'll just be body text `res.text()`.  I think you should check the content type and wrap the `await res.xyz()` method in a try/catch with a fallback to sending the statusText or statusCode.

Reply:
Do you have an example of what would returned in a BadRequest as json?

Reply:
No and I wouldn't expect it to be JSON at all.  But you can make a sample endpoint that throws different errors and test them.  Make one that throws BadRequest with a message, one that throws a blank one, one that just divides by 0, etc.  Try out a few thrown exception cases and then you can see if the change to the `!res.ok` error condition of our fetch method that you are making will work in different cases.